File: selection_sort/selection_sort.dart
void main(List<String> args) {
  print(selectionSort([5, 2, 4, 6, 1, 24, 15, 6]));
}

List<int> selectionSort(List<int> arr) {
  final List<int> newArr = List.empty(growable: true);

  final arrLenght = arr.length;

  for (var i = 0; i < arrLenght; i++) {
    final smallest = findSmallest(arr);
    newArr.add(arr.removeAt(smallest));
  }

  return newArr;
}

int findSmallest(List<int> arr) {
  int smallest = arr[0];
  int smallestIndex = 0;

  for (var i = 1; i < arr.length; i++) {
    if (arr[i] < smallest) {
      smallest = arr[i];
      smallestIndex = i;
    }
  }
  return smallestIndex;
}

File: selection_sort/selection_sort.py
def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0 

    for i in range(1, len(arr)):
        if arr[i] < smallest:
            smallest = arr[i]
            smallest_index = i
    return smallest_index        

def selectionSort(arr):
    newArr = []
    for i in range ( len(arr)):
        smallest = findSmallest(arr)
        newArr.append(arr.pop(smallest))        #The list pop in Python returns the popped value, or in other words, the item present at the specified index value.
    return newArr


print(selectionSort([5,6,2,10,23]))